### ISSUE TITLE
Nantex gets action after bumping

### ISSUE THREAD

**Issue by cm5089** [Sandrem/FlyCasual#2083]
Sun Fac got an action after bumping but using the tractor-roll/ensnare ability. Rotate arc/tractor should activate but the action should still be skipped.

**Comment by bitsai** [comment]
I don't understand; Ensnare triggers at the end of the Activation Phase, after the Perform Action step.  So if the bump was created using Ensnare, that means the bump happened after the Perform Action step for Sun Fac and so would not/cannot retroactively deny him his action.

Do you mind providing a screenshot or detailed steps to reproduce the issue you're seeing?

**Comment by cm5089** [comment]
Ignore ensnare, I'm referring to the ship ability which does not require a fully executed manoeuvre. The Nantex bumps, but it still gets to rotate it's arc and gain a tractor token (and therefore moves). The bug happens when the Nantex still gets to perform its action - probably because it no longer counts in the program as having bumped after moving out of range 0 (speculation, I don't know the logic you are using). It should have lost the action when it bumped but still been able to rotate it's arc/tractor.

**Comment by bitsai** [comment]
Ah, I think I understand now.  I was able to reproduce this in the game:

- Nantex bumps
- uses ship ability (Pinpoint Tractor Array) to gain a tractor token and rotate arc
- after moving away from bump due to tractor, Nantex gets to perform action as usual

I did a quick search on the FFG forums and the latest rules reference, you're right, Nantex's Perform Action step should have been skipped due to the initial bump.  I'll try to think a clean way to fix this soon.  Thank you for the helpful clarifications!
